### DISCUSSION TITLE
Conditional Fields

### DISCUSSION THREAD

**Discussion by cpt-n3mo** [keystonejs/keystone#7445] [Feature requests]
Good day.

is there a way to create a conditional field..?

for instance when one would have 3 fields `hesButton: checkbox()` `buttonLink:text()` `buttonText:text()`
is it possible to only render `buttonText` `buttonLink`  in admin ui,  when `hesButton` is true?
i noticed i can set some things with `ui.createView.fieldMode`
but i cant seem to find out a way to get to the checkbox field from context.

it would  be nice if one could get the fields within the list. so it could be evaluated using a simple if statement 

kind regards.

**Comment by raveling** [comment]
@cpt-n3mo we don't have conditional fields in place at the moment, but it is on the roadmap.

**Comment by jj-matos** [comment]
I see it is on the roadmap "Conditional fields - Dynamically showing fields based on the value of other fields". But is there a way at the moment to show a field based on its own value?

Something like this:

```
endDate: customCalendarDay({
  label:"End Date",
  ui: {
    listView: {
      fieldMode: async ({ context }) => {
        try {                                     
          const { event } = await context.query.Event.findOne({
            where: { endDate: endDate.toString() },
            query: 'event { endDate }',
          })      
          return endDate !== null ?  'read' : 'hidden'
        } catch (error) {
          console.error(error)
        }
      },
    },          
  }
}),
  ```
  
With this code I get the error: Cannot find name 'endDate'.
  
Went through all the context object, but couldn't find it. Is it possible?
